Hi, Do you get the error following when execute the launch?
```bash
$ ros2 launch azure_kinect_ros_driver driver.launch.py

[INFO] [launch]: All log files can be found below /root/.ros/log/2022-11-03-07-52-45-234984-jun-15580
[INFO] [launch]: Default logging verbosity is set to INFO
Robot description xacro_file : /root/ws_moveit/install/azure_kinect_ros_driver/share/azure_kinect_ros_driver/urdf/azure_kinect.urdf.xacro
Robot description urdf_path : /root/ws_moveit/install/azure_kinect_ros_driver/share/azure_kinect_ros_driver/urdf/azure_kinect.urdf
[ERROR] [launch]: Caught exception in launch (see debug for traceback): package 'azure_kinect_ros_driver' found at '/root/ws_moveit/install/azure_kinect_ros_driver', but libexec directory '/root/ws_moveit/install/azure_kinect_ros_driver/lib/azure_kinect_ros_driver' does not exist
```

Finally, it can be executed after revised CMakeLists.txt 
```cmake
install(TARGETS ${PROJECT_NAME}_node
  RUNTIME DESTINATION lib/${PROJECT_NAME}/
)
```

--- a/package.xml
+++ b/package.xml
@@ -18,6 +18,7 @@
   <depend>std_msgs</depend>
   <depend>sensor_msgs</depend>
   <depend>visualization_msgs</depend>
+  <depend>joint_state_publisher</depend>
   <depend>image_transport</depend>
   <depend>tf2</depend>
   <depend>tf2_ros</depend>

--- a/src/k4a_ros_device.cpp
+++ b/src/k4a_ros_device.cpp
@@ -254,14 +254,6 @@ K4AROSDevice::K4AROSDevice()
   depth_raw_publisher_ = image_transport_->advertise("depth/image_raw", 1, true);
   depth_raw_camerainfo_publisher_ = this->create_publisher<CameraInfo>("depth/camera_info", 1);
 
-  if (params_.depth_unit == sensor_msgs::image_encodings::TYPE_16UC1) {
-    // set lowest PNG compression for maximum FPS
-    this->set_parameter({depth_raw_topic + compressed_format, "png"});
-    this->set_parameter({depth_raw_topic + compressed_png_level, 1});
-    this->set_parameter({depth_rect_topic + compressed_format, "png"});
-    this->set_parameter({depth_rect_topic + compressed_png_level, 1});
-  }
-
   depth_raw_publisher_ = image_transport_->advertise(depth_raw_topic, 1, true);
   depth_raw_camerainfo_publisher_ = this->create_publisher<CameraInfo>("depth/camera_info", 1);
 

